backend/remote-state/swift/client.go

```diff
@@ -0,0 +1,116 @@
+package swift
+
+import (
+	"bytes"
+	"crypto/md5"
+	"log"
+	"os"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/containers"
+	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects"
+
+	"github.com/hashicorp/terraform/state/remote"
+)
+
+const (
+	TFSTATE_NAME      = "tfstate.tf"
+	TFSTATE_LOCK_NAME = "tfstate.lock"
+)
+
+// RemoteClient implements the Client interface for an Openstack Swift server.
+type RemoteClient struct {
+	client      *gophercloud.ServiceClient
+	path        string
+	archive     bool
+	archivepath string
+	expireSecs  int
+}
+
+func (c *RemoteClient) Get() (*remote.Payload, error) {
+	log.Printf("[DEBUG] Getting object %s at path %s", TFSTATE_NAME, c.path)
+	result := objects.Download(c.client, c.path, TFSTATE_NAME, nil)
+
+	// Extract any errors from result
+	_, err := result.Extract()
+
+	// 404 response is to be expected if the object doesn't already exist!
+	if _, ok := err.(gophercloud.ErrDefault404); ok {
+		log.Println("[DEBUG] Object doesn't exist to download.")
+		return nil, nil
+	}
+
+	bytes, err := result.ExtractContent()
+	if err != nil {
+		return nil, err
+	}
+
+	hash := md5.Sum(bytes)
+	payload := &remote.Payload{
+		Data: bytes,
+		MD5:  hash[:md5.Size],
+	}
+
+	return payload, nil
+}
+
+func (c *RemoteClient) Put(data []byte) error {
+	if err := c.ensureContainerExists(); err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Putting object %s at path %s", TFSTATE_NAME, c.path)
+	reader := bytes.NewReader(data)
+	createOpts := objects.CreateOpts{
+		Content: reader,
+	}
+
+	if c.expireSecs != 0 {
+		log.Printf("[DEBUG] ExpireSecs = %d", c.expireSecs)
+		createOpts.DeleteAfter = c.expireSecs
+	}
+
+	result := objects.Create(c.client, c.path, TFSTATE_NAME, createOpts)
+
+	return result.Err
+}
+
+func (c *RemoteClient) Delete() error {
+	log.Printf("[DEBUG] Deleting object %s at path %s", TFSTATE_NAME, c.path)
+	result := objects.Delete(c.client, c.path, TFSTATE_NAME, nil)
+	return result.Err
+}
+
+func (c *RemoteClient) ensureContainerExists() error {
+	log.Println("[DEBUG] Got to ensureContainerExists...")
```

Comment:
Yeh, the log lines were useful to a point... Agree that line is superfluous though...